worker.go

```diff
@@ -43,17 +44,49 @@ func NewWorkerMetrics() WorkerMetrics {
 	}
 }
 
+// Create an entry in wm for the given metrickey, if it does not already exist.
+// Returns true if the metric entry was created and false otherwise.
+func (wm WorkerMetrics) Upsert(mk MetricKey, tags []string) bool {
```

Comment:
Rather than returning a `bool` — which nothing seems to use fwict — how about returning the `Sampler` instance that was found or created. Then we can avoid the `switch` in `ProcessMetric` and just do `<returned value>.Sample(m.Value, m.SampleRate)`? That, however, may require some sort of casting and not work. My gofu isn't strong enough without trying it… but maybe that comment will be useful? :)